accounts/backend.py

```diff
@@ -0,0 +1,42 @@
+from django.conf import settings
+from social.backends.oauth import BaseOAuth2
+from urllib.parse import urlencode
+import json
+import urllib
+
+
+class OpenStaxOAuth2(BaseOAuth2):
+    """openstax OAuth authentication backend"""
+    name = 'openstax'
+    AUTHORIZATION_URL = settings.AUTHORIZATION_URL 
+    ACCESS_TOKEN_URL = settings.ACCESS_TOKEN_URL 
+    USER_QUERY = settings.USER_QUERY
+    REQUEST_TOKEN_METHOD = 'POST'
+    ACCESS_TOKEN_METHOD = 'POST'
+    SCOPE_SEPARATOR = ','
+    EXTRA_DATA = [
+        ('id', 'id'),
+    ]
+
+    def get_user_details(self, response):
+        """Return user details from openstax account's"""
+        return {'id': response.get('id'),
+                'username': response.get('username'),
+                'first_name':response.get('first_name'),
+                'last_name':response.get('last_name'),
+                'full_name': response.get('full_name')}
+
```

Comment:
If the user logs in with Twitter, we do not get an email address. I'm 
not sure having an email is a requirement. All emails needed for 
workflow will be sent through Salesforce.

On 01/22/2016 12:48 PM, Michael Harrison wrote:
>
> In accounts/backend.py 
> <https://github.com/openstax/openstax-cms/pull/54#discussion_r50574488>:
>
> > +    USER_QUERY = settings.USER_QUERY
> > +    REQUEST_TOKEN_METHOD = 'POST'
> > +    ACCESS_TOKEN_METHOD = 'POST'
> > +    SCOPE_SEPARATOR = ','
> > +    EXTRA_DATA = [
> > +        ('id', 'id'),
> > +    ]
> > +
> > +    def get_user_details(self, response):
> > +        """Return user details from openstax account's"""
> > +        return {'id': response.get('id'),
> > +                'username': response.get('username'),
> > +                'first_name':response.get('first_name'),
> > +                'last_name':response.get('last_name'),
> > +                'full_name': response.get('full_name')}
> > +
>
> Is the email available from the service? I think this was a requirement.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/openstax/openstax-cms/pull/54/files#r50574488>.
>
> !DSPAM:1106,56a2799971001228515347!